### ISSUE TITLE
Webpage and Python char not being shown on some Linux distributions

### ISSUE THREAD

**Issue by donc3** [GoldenCheetah/GoldenCheetah#4196]
Webpage and python charts are not being shown when `GC.webpage` is used. Although the temporary HTML file is being generated. Tried checking and unchecking to show web chart but still not doing nothing.
No logs are shown in --debug option.

* Version :

![image](https://user-images.githubusercontent.com/560524/159118261-93b75104-0bb5-4df8-9356-897dfc4c2f59.png)

* Demo video:

![Peek 19-03-2022 11-49](https://user-images.githubusercontent.com/560524/159118183-cf7dd17f-8fb2-4935-bd2b-d076b549fdbb.gif)

* HTML file:

![image](https://user-images.githubusercontent.com/560524/159118335-2de367cd-28d1-4c54-8caf-cb05cf71dc1b.png)

**Comment by amtriathlon** [comment]
The same issue has been reported at the users forum on other rolling release Linux distributions s.t. Arch Linux and OpenSUSE Tumbleweed.

Apparently a recent change in the OS is provoking qtwebengineproc to crash when Qt 5.14.2 (or previous, it also happens with GCv3.5) WebEngine is used to render webpages. You can check this using journalctl.

It is likely building using Qt 5.15.2 fix this issue, but we need testing.

**Comment by amtriathlon** [comment]
Building Qt5.15.2 doesn't fix this issue according to feedback in users forum, QtWebEngineProc keeps crashing on Arch Linux and derivatives s.t. Manjaro.

**Comment by srwareham** [comment]
I can confirm I'm having the same issue on Arch. All of the charts from CloudDB that use plotly don't work on my system. However, those that use qt charts (e.g., GC.setChart) all work fine.

Attempting to open even the simplest hello world html using `GC.webpage(PATH)` returns a 1 exit code with no stack trace or anything I can seem to find.


```
$ GC.version()
V3.6-DEV2009
```

```
$ uname -a
Linux Arch 5.17.4-arch1-1 #1 SMP PREEMPT Wed, 20 Apr 2022 18:29:28 +0000 x86_64 GNU/Linux
```

Also using `qt5-base 5.15.3+kde+r145-1` as the system's QT version

Happy to help with any debugging/testing if people have ideas.

**Comment by amtriathlon** [comment]
What we know is QtWebEngineProc (the Qt WebEngine process rendering web pages) crashes on these distributions and building with Qt 5.15.2 doesn’t fix the issue.
OTOH the Qt version installed in the system is not relevant, since the AppImage includes its own version. If someone is Interested in playing with that, the way would be to extract the AppImage to the file system and remove/replace Qt components.

**Comment by srwareham** [comment]
Thank you, @amtriathlon! By removing all of the QT libraries from the AppImage I'm now able to get all charts to work on Arch Linux!


Full steps to replicate working charts

1. Download GoldenCheetah AppImage (i used 3.6-DEV2009)
2. Run  `./GoldenCheetah_v3.6-DEV2009_x64.AppImage --appimage-mount` to copy the GC AppImage files to disk (might need to install appimagetool first, not sure). Default mount point should make the files accessible somewhere in /tmp
3. Navigate to the relevant folder.
4. Find every file that contains Qt or qt and move them to placeholder directory or delete them
5. Install any QT libraries not currently on your system. For me, I had to run `pikaur -S qt5-connectivity qt5-bluetooth qt5-serialport qt5-charts`. If you're not sure, just continue to the step below and you'll get an error message telling you exactly what libraries you're missing.
6. Within the directory containing the AppImage files, run GoldenCheetah with `./AppRun`
7. If there are any more missing libraries, install them manually with pacman or the AUR
8. Done!

On my local system  libQt5Charts.so points to libQt5Charts.so.5.15.3. I'd guess a version bump of packaging qt 5.15 might resolve the issue. I'm not familiar with AppImage packaging and am not quite sure how to test myself. I assume there are constraints that might be blocking such a version bump?

```

**Comment by amtriathlon** [comment]
> I assume there are constraints that might be blocking such a version bump?

Qt 5.15.2 is the last Qt 5 version readily available for FOSS projects, AFAIK.

**Comment by MeisterP** [comment]
@srwareham  Could you try whether running with `QTWEBENGINE_DISABLE_SANDBOX` fixes your issue?

`QTWEBENGINE_DISABLE_SANDBOX=1 ./GoldenCheetah_v3.6-DEV_x64.AppImage`

**Comment by srwareham** [comment]
@MeisterP I can confirm that works! All types of charts work on the stock-standard AppImage when passing the flags you suggest. Thank you!

**Comment by MeisterP** [comment]
If I understand it correctly, it's an issue with qtwebengine (chromium) and glibc 2.34

https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1939993
looks like the fix is not part of beineri/opt-qt-5.14.2-bionic ppa thought
